src/domain/usecases/CopyDataSetsOUUserCase.ts

```diff
@@ -0,0 +1,96 @@
+import _ from "lodash";
+import { DataSetsRepository } from "domain/repositories/DataSetsRepository";
+import { PostOptions, Ref } from "types/d2-api";
+import { DataSet } from "domain/entities/DataSet";
+import { OUCopyResult } from "domain/entities/OUCopyResult";
+
+export class CopyDataSetsOUUserCase {
+    constructor(
+        private dataSetsRepository: DataSetsRepository
+    ) { }
+
+    async execute(options: {
+        originDataset: string;
+        destinationDatasets: string[];
+        replace: boolean | undefined;
+    }): Promise<OUCopyResult> {
+        const { originDataset, destinationDatasets, replace } = options;
```

Comment:
It's idiomatic to set the default values here: `...., replace = false } = options`